3.0/m_conn_banner.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2012 Attila Molnar <attilamolnar@hush.com>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/// $ModAuthor: Attila Molnar
+/// $ModAuthorMail: attilamolnar@hush.com
+/// $ModConfig: <connbanner text="Banner text goes here.">
+/// $ModDepends: core 3.0
+/// $ModDesc: Displays a static text to every connecting user before registration
+
+
+#include "inspircd.h"
+
+class ModuleConnBanner : public Module
+{
+	LocalIntExt init;
+	std::string text;
+
+ public:
+	ModuleConnBanner()
+		: init("user_init", ExtensionItem::EXT_USER, this)
+	{
+	}
+
+	void ReadConfig(ConfigStatus&) CXX11_OVERRIDE
+	{
+		text = ServerInstance->Config->ConfValue("connbanner")->getString("text");
+	}
+
+	void OnSetUserIP(LocalUser* user) CXX11_OVERRIDE
+	{
+		if (init.get(user))
```

Comment:
You unset this on `OnUserConnect` but there is no guarantee that a user's IP will not change after registration (e.g. via the IRCv3 resume spec).

Reply:
I'm starting to like the thought of a `OnUserPostInit` or `OnUserPreRegister` :smile:
I'd rather not leave ExtItems if I don't need to so if I check the registration state is none as well as the ExtItem it should cover all. The ExtItem will remain until connected and by then the registration state is definitely not none.